chia/wallet/dao_wallet/dao_wallet.py

```diff
@@ -0,0 +1,2268 @@
+                int_from_bytes(cond.at("rrf").as_atom()) == 1
+            ):
+                cat_tail_hash = bytes32(cond.at("rrrff").as_atom())
+                break
+        assert cat_tail_hash
+
+        cat_wallet: Optional[CATWallet] = None
+
+        # Get or create a cat wallet
+        for wallet_id in self.wallet_state_manager.wallets:
+            wallet = self.wallet_state_manager.wallets[wallet_id]
+            if wallet.type() == WalletType.CAT:  # pragma: no cover
+                assert isinstance(wallet, CATWallet)
+                if wallet.cat_info.limitations_program_hash == cat_tail_hash:
+                    cat_wallet = wallet
+                    break
+        else:
+            # Didn't find a cat wallet, so create one
+            cat_wallet = await CATWallet.get_or_create_wallet_for_cat(
+                self.wallet_state_manager, self.standard_wallet, cat_tail_hash.hex()
+            )
+
+        assert cat_wallet is not None
+        cat_wallet_id = cat_wallet.wallet_info.id
+        dao_info = dataclasses.replace(
+            self.dao_info,
+            cat_wallet_id=uint32(cat_wallet_id),
+            dao_cat_wallet_id=uint32(0),
+            current_treasury_coin=child_coin,
+            current_treasury_innerpuz=current_inner_puz,
+        )
+        await self.save_info(dao_info)
+
+        future_parent = LineageProof(
+            child_coin.parent_coin_info,
+            dao_info.current_treasury_innerpuz.get_tree_hash(),
+            uint64(child_coin.amount),
+        )
+        await self.add_parent(child_coin.name(), future_parent)
+        assert self.dao_info.parent_info is not None
+
+        # get existing xch funds for treasury
+        xch_funds_puzhash = get_p2_singleton_puzhash(self.dao_info.treasury_id, asset_id=None)
+        await self.wallet_state_manager.add_interested_puzzle_hashes([xch_funds_puzhash], [self.id()])
+        await self.wallet_state_manager.add_interested_puzzle_hashes([self.dao_info.treasury_id], [self.id()])
+        await self.wallet_state_manager.add_interested_puzzle_hashes(
+            [self.dao_info.current_treasury_coin.puzzle_hash], [self.id()]
+        )
+
+        # Resync the wallet from when the treasury was created to get the existing funds
+        # TODO: Maybe split this out as an option for users since it may be slow?
+        if not wallet_node.is_trusted(peer):
+            # Untrusted nodes won't automatically send us the history of all the treasury and proposal coins,
+            # so we have to request them via sync_from_untrusted_close_to_peak
+            request = RequestBlockHeader(children_state.created_height)
+            response: Optional[RespondBlockHeader] = await peer.call_api(FullNodeAPI.request_block_header, request)
+            await wallet_node.sync_from_untrusted_close_to_peak(response.header_block, peer)
+
+        return
+
+    async def create_tandem_xch_tx(
```

Comment:
there's a `Wallet.create_tandem_xch_tx` function now that you could probably use.

Reply:
replaced the old tandem_tx with this one